examples/vision/reptile.py

```diff
@@ -0,0 +1,248 @@
+        if split:
+            return dataset, test_images, test_labels
+        return dataset
+
+
+import urllib3
+
+urllib3.disable_warnings()
+train_dataset = Dataset(training=True)
+test_dataset = Dataset(training=False)
+
+"""
+## Build the Model
+"""
+
+
+class ModelLayer(layers.Layer):
+    def __init__(self, **kwargs):
+        super(ModelLayer, self).__init__(**kwargs)
+        self.conv = layers.Conv2D(filters=64, kernel_size=3, strides=2, padding="same")
+        self.bnorm = layers.BatchNormalization()
+        self.activation = layers.ReLU()
+
+    def call(self, inputs):
+        x = self.conv(inputs)
+        x = self.bnorm(x)
+        return self.activation(x)
+
+
+inputs = layers.Input(shape=(28, 28, 1))
+layer1 = ModelLayer()(inputs)
+layer2 = ModelLayer()(layer1)
+layer3 = ModelLayer()(layer2)
+layer4 = ModelLayer()(layer3)
+flatten = layers.Flatten()(layer4)
+dense = layers.Dense(classes, activation="softmax")(flatten)
+# Log softmax
+outputs = tf.math.log(layers.Softmax()(dense))
+model = keras.Model(inputs=inputs, outputs=dense)
+model.compile()
+optimizer = keras.optimizers.SGD(learning_rate=learning_rate)
+
+"""
+## Training Loop
+"""
+
+training = []
+testing = []
+for meta_iter in range(meta_iters):
+    frac_done = meta_iter / meta_iters
+    cur_meta_step_size = (1 - frac_done) * meta_step_size
+    # Temporarily save the weights from the model.
+    old_vars = model.get_weights()
+    # Get a sample from the full dataset.
+    mini_dataset = train_dataset.get_mini_dataset(
+        inner_batch_size, inner_iters, train_shots, classes
+    )
+    for images, labels in mini_dataset:
+        with tf.GradientTape() as tape:
+            logits = model(images)
+            loss = keras.losses.categorical_crossentropy(
+                tf.one_hot(labels, depth=classes), logits
+            )
+        grads = tape.gradient(loss, model.trainable_weights)
+        optimizer.apply_gradients(zip(grads, model.trainable_weights))
+    new_vars = model.get_weights()
+    # Perform SGD for the meta step.
+    for var in range(len(new_vars)):
+        new_vars[var] = old_vars[var] + (
+            (new_vars[var] - old_vars[var]) * cur_meta_step_size
+        )
+    # After the meta-learning step, reload the newly trained weights into the model.
+    model.set_weights(new_vars)
+    # Evaluation loop
+    if meta_iter % eval_interval == 0:
+        accuracies = []
+        for dataset in (train_dataset, test_dataset):
+            # Sample a mini dataset from the full dataset.
+            train_set, test_images, test_labels = dataset.get_mini_dataset(
+                eval_batch_size, eval_iters, shots, classes, split=True
+            )
+            old_vars = model.get_weights()
+            # Train on the samples and get the resulting accuracies.
+            for images, labels in train_set:
+                with tf.GradientTape() as tape:
+                    logits = model(images)
```

Comment:
If these are supposed to be logits, then there should be no softmax and you should use `from_logits=True`